docs/site/Parsing-requests.md

```diff
@@ -234,6 +234,76 @@ class Product extends Entity {
 A full list of options & usage scenarios could be found in the
 [documentation of AJV errors](https://github.com/epoberezkin/ajv-errors).
 
+##### Custom AJV Formats
+
+AJV Custom Formats can be configured by binding the value
+`validation.ajvCustomFormats` to `RestBindings.REQUEST_BODY_PARSER_OPTIONS`
+('rest.requestBodyParserOptions').
+
+For example,
+
+```ts
+server.bind(RestBindings.REQUEST_BODY_PARSER_OPTIONS).to({
+  validation: {
+    ajvCustomFormats: {
+      'only-alphanumeric': /^[a-zA-Z0-9 ]+$/.source,
+      'product-code': {
+        async: true,
+        validate: async (code: string) => {
+          return /^[a-zA-Z]{3}-\d{3}$/.test(code)
+            ? Promise.resolve(true)
+            : Promise.resolve(false);
+        },
+      },
+    },
+  },
+});
```

Comment:
There's already a `pattern` key under the  `@property`. This  would, AFAIK, reflect under the OpenAPI 3 schema, while AJV custom formats won't as it's a function. The pattern can also be stored inside a variable for re-use, similar to what AJV custom formats is doing.

Is there a benefit to using AJV's Custom Formats?

Reply:
The example I have used is a simpler one, however, use cases are there.

A `format` for ajv can do more than just pattern match.

Example from ajv library that does the async process for a format.

https://github.com/epoberezkin/ajv/blob/0020556493f3d443002596697768f3f047782198/spec/async_validate.spec.js#L379-L394

One more: https://runkit.com/esp/ajv-asynchronous-validation

Reply:
Ah noted;